packages/ember-glimmer/tests/integration/content-test.js

```diff
@@ -584,4 +585,236 @@ moduleFor('Dynamic content tests (integration)', class extends RenderingTest {
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('fizz') } });
+
+    this.runTask(() => set(this.context, 'model', { foo: 'foo' }));
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo') } });
+  }
+
+  ['@test quoted class property']() {
+    this.render('<div class="{{model.foo}}">hello</div>', {
+      model: {
+        foo: 'foo'
+      }
+    });
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo') } });
+
+    this.runTask(() => this.rerender());
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo') } });
+
+    this.runTask(() => set(this.context, 'model.foo', 'fizz'));
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('fizz') } });
+
+    this.runTask(() => set(this.context, 'model', { foo: 'foo' }));
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo') } });
+  }
+
+  ['@test quoted class property can contain multiple classes']() {
+    this.render('<div class="{{model.classes}}">hello</div>', {
+      model: {
+        classes: 'foo bar baz'
+      }
+    });
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo bar baz') } });
+
+    this.runTask(() => this.rerender());
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo bar baz') } });
+
+    this.runTask(() => set(this.context, 'model.classes', 'fizz bizz'));
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('fizz bizz') } });
+
+    this.runTask(() => set(this.context, 'model', { classes: 'foo bar baz' }));
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo bar baz') } });
+  }
+
+  ['@test class attribute concats bound values']() {
+    this.render('<div class="{{model.foo}} {{model.bar}} {{model.bizz}}">hello</div>', {
+      model: {
+        foo: 'foo',
+        bar: 'bar',
+        bizz: 'bizz'
+      }
+    });
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo bar bizz') } });
+
+    this.runTask(() => this.rerender());
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo bar bizz') } });
+
+
+    this.runTask(() => set(this.context, 'model.foo', 'fizz'));
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('fizz bar bizz') } });
+
+    this.runTask(() => set(this.context, 'model.bar', null));
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('fizz  bizz') } });
```

Comment:
I see what's going on now. The spirit of the `classes` helper is to ignore spaces and ordering – but we currently have a bug that sometimes give you extra `""`. I think we should fix the helper to split with a whitespace regex (instead of a single space character) and not worry about it here. If for some reason we do want to test the actual output (I don't think we want to), character by character, we can always just pass a string here to do an exact match instead of using the `classes` helper.